src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -836,7 +836,7 @@ private void EmbedSourceLink(Stream stream)
         }
 
         ///<summary>The version of the compilation options schema to be written to the PDB.</summary>
-        public const int CompilationOptionsSchemaVersion = 1;
+        public const int CompilationOptionsSchemaVersion = 2;
```

Comment:
What are the implications of this change? Is there a test that could demonstrate the effect of changing the version?

Reply:
> What are the implications of this change?

It informs tools that read the options from the PDB know if it was created from Roslyn 3.9/VS 16.9 (schema version 1) or Roslyn 3.10/VS 16.10 (schema version 2). I don't know the full reasons *why* this is useful but I could imagine having different "compatibility" behaviors as, for example, the expected fields or the format of values within the fields changes between schema versions.

Now that I see this I wonder why we didn't just make it the same as the VS or Roslyn major+minor version number.. should we change that? Or is it too late to adopt a new convention? :/

> Is there a test that could demonstrate the effect of changing the version?

Reply:
The SymbolValidator in NuGet Package Explorer, which drives the data we're collecting on ecosystem health, reads this value to determine if the package has a new enough compiler to be reproducible. 

It's also there to future-proof, in case we need to change what reproducible means. We can say we know that for version 2, do this, but for version 3, some other thing needs to happen as well.

I think it's not tied to the Roslyn version because in theory, this should rev infrequently as it's a product of what we need to reproduce builds, not what Roslyn's overall feature set has (which is ever growing).

Reply:
We check the `"version"` field in the tests, but since the test references the same constant, there is no need to update the test baselines..do you see any need to make a change to this @cston?

https://github.com/dotnet/roslyn/blob/d0e64666ce95ac6279c1ae346d50446e976060e2/src/Compilers/Test/Core/PDB/DeterministicBuildCompilationTestHelpers.cs#L49-L49

Reply:
It seems reasonable to me to check explicitly for `2` in `DeterministicBuildCompilationTestHelpers`, so that some test catches changes to the `CompilationOptionsSchemaVersion`.